compiler_opt/rl/feature_ops_test.py

```diff
@@ -43,35 +43,25 @@ def setUp(self):
     self._quantile_file_dir = os.path.join(constant.BASE_DIR, 'testdata')
     super().setUp()
 
-  def _create_temp_vocab_file(self, vocab_data):
-    """Helper method to create a temporary vocab file with given data."""
-    with tempfile.NamedTemporaryFile(
-        mode='w', suffix='.json', delete=False) as f:
-      if isinstance(vocab_data, str):
-        f.write(vocab_data)  # For invalid JSON
-      else:
-        json.dump(vocab_data, f)
-      return f.name
-
   def _test_vocab_file_dimensions(self, vocab_data, expected_dimensions):
     """Helper method to test vocab file dimension reading."""
-    temp_file = self._create_temp_vocab_file(vocab_data)
-    try:
-      dimensions = feature_ops.get_ir2vec_dimensions_from_vocab_file(temp_file)
+    with tempfile.NamedTemporaryFile(
+        mode='w', suffix='.json', delete=True) as f:
+      json.dump(vocab_data, f)
+      f.flush()
```

Comment:
Are the flush calls just precautionary or did you actually run into problems?

Either way, we probably want to keep them because I don't think `json.dump` makes any guarantees.

Reply:
Yes, without flush we would end up with decode errors.